source/Octopus.Cli/Commands/Package/PushCommand.cs

```diff
@@ -24,8 +24,9 @@ public PushCommand(IOctopusAsyncRepositoryFactory repositoryFactory, IOctopusFil
             var options = Options.For("Package pushing");
             options.Add("package=", "Package file to push. Specify multiple packages by specifying this argument multiple times: \n--package package1 --package package2", 
                 package => Packages.Add(EnsurePackageExists(fileSystem, package)));
-            options.Add("replace-existing", "If the package already exists in the repository, the default behavior is to reject the new package being pushed. You can pass this flag to overwrite the existing package.", 
-                replace => ReplaceExisting = true);
+            options.Add("overwrite-mode=", "If the package already exists in the repository, the default behavior is to reject the new package being pushed (FailIfExists). You can use the overwrite mode to OverwriteExisting or IgnoreIfExists.", mode => OverwriteMode = (OverwriteMode)Enum.Parse(typeof(OverwriteMode), mode));
+            options.Add("replace-existing", "If the package already exists in the repository, the default behavior is to reject the new package being pushed. You can pass this flag to overwrite the existing package. This flag may be deprecated in a future version, passing it is the same as using the OverwriteExisting overwrite-mode.", 
```

Comment:
This feels clearer to me with a semicolon:
```suggestion
            options.Add("replace-existing", "If the package already exists in the repository, the default behavior is to reject the new package being pushed. You can pass this flag to overwrite the existing package. This flag may be deprecated in a future version; passing it is the same as using the OverwriteExisting overwrite-mode.", 
```